### ISSUE TITLE
ProperEscapingFunctionSniff doesn't catch misuse of wp_json_encode

### ISSUE THREAD

**Issue by iandunn** [Automattic/VIP-Coding-Standards#876]
## Bug Description

[ProperEscapingFunctionSniff](https://github.com/Automattic/VIP-Coding-Standards/blob/54ff2f355aa6bdf5d987636a8e4b817d91e32d5d/WordPressVIPMinimum/Sniffs/Security/ProperEscapingFunctionSniff.php#L20) is a great addition to WPCS. It isn't comprehensive, though.

One specific example that I ran into is that it won't catch improper use of `wp_json_encode()` like it does with other functions.

## Minimal Code Snippet

```html
<!-- correctly flags esc_html as the wrong escaping function in this context -->
<img src=a onerror="<?php echo esc_html( $foo ); ?>" />

<!-- fails to catch that this is the wrong escaping function -->
<img src=a onerror="<?php echo wp_json_encode( $foo ); ?>" />
```

## Error Code

`WordPressVIPMinimum.Security.ProperEscapingFunction`
